### ISSUE TITLE
Should major version be increased every time a new rule is added?

### ISSUE THREAD

**Issue by alefranz** [dotnet/format#292]
Recently released version 3.1 is a minor update as it adds new features.
However new features means handling new rules, for example it now handle the new line at end of file rule.
This mean that if you are relying on dotnet format on your build pipeline, and some of your files are not following this previously ignored rule, on update your build will fail, which is a bit nasty especially if you build install the latest version of the tool.
So from a consumer point of you, this actually behave like a breaking change, so maybe it deserve a major version update.
Thoughts?

**Comment by jmarolf** [comment]
Yeah, I think this is correct.  @JoeRobich 	well want to rev the version to 4.0 once all the new rules are in. Hopefully once we are at feature parity with roslyn we won't need to rev the main version number

**Comment by sharwell** [comment]
I don't agree with this at all. Minor versions should be allowed to add new features, especially if those features can be turned off. If a project is using dotnet-format for validation during a build (where this scenario could break them), the _consumer_ must take one of the following actions:

1. Lock the version of dotnet-format (disable auto-upgrade)
2. Accept the breaking changes

**Comment by alefranz** [comment]
> 2. Accept the breaking changes
The fact that it is a breaking change is what matter here! :) If this tool follows SemVer it would mean a new major version.
This would allow a better flexibility for consumers in locking a version range (essentially pinning the major version) while still allow who prefers to be on latest to do so.

**Comment by sharwell** [comment]
@alefranz if we were talking about the public API, I'd be more inclined to agree. However, it turns out that most changes made to this package could be considered "breaking" on some definition. The one case where it's reasonably clear that a breaking change does not occur is the addition of a new rule, especially if the command line offers the ability run a set of rules by name. When a new rule is added, the new rule would not actually run or be used until the client updates their command line invocation to include the new rule.
